pkgs/kde/third-party/wallpaper-engine-plugin/default.nix

```diff
@@ -0,0 +1,60 @@
+{
+  extra-cmake-modules,
+  fetchFromGitHub,
+  full,
+  kpackage,
+  libplasma,
+  lib,
+  lz4,
+  mkKdeDerivation,
+  mpv,
+  pkg-config,
+  python3,
+}:
+mkKdeDerivation {
+  pname = "wallpaper-engine-kde-plugin";
+  version = "qt6";
+
+  src = fetchFromGitHub {
+    owner = "catsout";
+    repo = "wallpaper-engine-kde-plugin";
+    rev = "1e604105c586c7938c5b2c19e3dc8677b2eb4bb4";
+    hash = "sha256-bKGQxyS8gUi+37lODLVHphMoQwLKZt/hpSjR5MN+5GA=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./nix-plugin.patch ];
+
+  extraBuildInputs = [
+    extra-cmake-modules
+    full
```

Comment:
What is "full"?

Reply:
I assume it's a collections of `kdePackages.qt*` packages that `cmake` can use during build. I'm not an expert in Qt, and I tried to isolate the individual packages used in build, but I have not been successful so far.

Reply:
That's a bad idea, you need to specify the exact components.

Reply:
It would be great if I could lend some help from the experts here.

At the same time there are some additional `kdePackages` required in the environment at runtime. I hope to package them together with the plugin as well, but I don't know how.

Reply:
This should have been resolved with the discussion below.